commit 7ec04db807c7e643056a0f95e42f972dbe7154ae (refs/changes/48/4648/1)
Author: Yuri Pankov <yuri.pankov@nexenta.com>
Date:   2018-08-07T17:46:01+00:00 (1 year, 2 months ago)
    
    9693 emulated NVMe controller on ESXi 6.7 fails to attach
    Reviewed by: Dan Fields <dan.fields@nexenta.com>
    Reviewed by: Rick McNeal <rick.mcneal@nexenta.com>
    Reviewed by: Robert Mustacchi <rm@joyent.com>
    Approved by: Hans Rosenfeld <rosenfeld@grumpf.hope-2000.org>

diff --git a/usr/src/uts/common/io/nvme/nvme.c b/usr/src/uts/common/io/nvme/nvme.c
index c6785b28d3..ec6d18ca91 100644
--- a/usr/src/uts/common/io/nvme/nvme.c
+++ b/usr/src/uts/common/io/nvme/nvme.c
@@ -83,8 +83,8 @@
  * NVMe devices can have multiple namespaces, each being a independent data
  * store. The driver supports multiple namespaces and creates a blkdev interface
  * for each namespace found. Namespaces can have various attributes to support
- * thin provisioning and protection information. This driver does not support
- * any of this and ignores namespaces that have these attributes.
+ * protection information. This driver does not support any of this and ignores
+ * namespaces that have these attributes.
  *
  * As of NVMe 1.1 namespaces can have an 64bit Extended Unique Identifier
  * (EUI64). This driver uses the EUI64 if present to generate the devid and
@@ -2223,16 +2223,13 @@ nvme_init_ns(nvme_t *nvme, int nsid)
 
 	/*
 	 * We currently don't support namespaces that use either:
-	 * - thin provisioning
 	 * - protection information
 	 * - illegal block size (< 512)
 	 */
-	if (idns->id_nsfeat.f_thin ||
-	    idns->id_dps.dp_pinfo) {
+	if (idns->id_dps.dp_pinfo) {
 		dev_err(nvme->n_dip, CE_WARN,
-		    "!ignoring namespace %d, unsupported features: "
-		    "thin = %d, pinfo = %d", nsid,
-		    idns->id_nsfeat.f_thin, idns->id_dps.dp_pinfo);
+		    "!ignoring namespace %d, unsupported feature: "
+		    "pinfo = %d", nsid, idns->id_dps.dp_pinfo);
 		ns->ns_ignore = B_TRUE;
 	} else if (ns->ns_block_size < 512) {
 		dev_err(nvme->n_dip, CE_WARN,
